control.go

```diff
@@ -282,6 +282,7 @@ func (c *controlConn) setupConn(conn *Conn) error {
 	}
 
 	if err := c.registerEvents(conn); err != nil {
+		c.session.logger.Printf("gocql: control unable to register events: %v\n", err)
```

Comment:
We should either handle the error (log or otherwise) or return it, not both. Otherwise we'd be handling the same error in multiple places. In this case it seems to me that we should wrap the error like `return fmt.Errorf("register events: %v", err)` instead of logging here and let the caller handle it, as the setup of the control connection failed and we don't continue setupConn with unregistered events.